File: proy/logico/programas/hanoi.pl
hanoi(1,A,B,C) :- write("Mueve del ",A," al ",C), nl.
hanoi(N,A,B,C) :- N>1, M is N-1, hanoi(M,A,C,B), hanoi(1,A,B,C), hanoi(M,B,A,C). 

File: proy/objetos/programas/hanoi.py
def hanoi(n, inc='1', temp='2', fin='3'):
  if n > 0:
      hanoi(n-1, inc, fin, temp)
      print('se mueve de torre', inc, 'a torre', fin)
      hanoi(n-1, temp, inc, fin)

hanoi(5)